client/src/route/auth/Signin.js

```diff
@@ -1,140 +1,135 @@
-/*
-src: https://github.com/mui/material-ui/blob/v5.10.0/docs/data/material/getting-started/templates/sign-in/SignIn.js
-*/
+/* eslint-disable no-console */
+/* eslint-disable max-len */ // disable eslint rule that prohibits lines over 100 chars.
+import './style.css';
+import React, { useEffect } from 'react';
+import {
+  loginWithGitlab, checkTokenValidity, resetToken, getGitLabAccessToken, fetchGitLabGraphQLQuery,
+} from './Functions';
 
-import * as React from 'react';
-import Avatar from '@mui/material/Avatar';
-import Button from '@mui/material/Button';
-import CssBaseline from '@mui/material/CssBaseline';
-import TextField from '@mui/material/TextField';
-import FormControlLabel from '@mui/material/FormControlLabel';
-import Checkbox from '@mui/material/Checkbox';
-import Toolbar from '@mui/material/Toolbar';
-import Link from '@mui/material/Link';
-import Grid from '@mui/material/Grid';
-import Box from '@mui/material/Box';
-import MuiAppBar from '@mui/material/AppBar';
-import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
-import Typography from '@mui/material/Typography';
-import Container from '@mui/material/Container';
-import { styled, createTheme, ThemeProvider } from '@mui/material/styles';
-import Footer from '../../page/Footer';
+// const REDIRECT_URI = 'http://localhost:4000/dashboard';
+// Define constants used throughout the application
+const CLIENT_ID = 'f05709a4c29009cc4bb443c2d13bcd4d9f457107495c5ebae5af8e8a1bed774d';
+const CLIENT_SECRET = 'e85ae3fb2e7b72d5f709ae2a0753e89e6ddd2beab9042150a0deb8880b1ab489';
+const REDIRECT_URI = 'http://localhost:4000/';
+const REQUESTED_SCOPES = 'read_user+profile+read_api+email+read_repository';
 
```

Comment:
To env file?

Reply:
Are you talking about the dev.js file from the path: client/config/dev.js or do you want us to make a new env file?

Reply:
To the new .env file of the react app.